lib/posix/Kconfig

```diff
@@ -39,6 +39,36 @@ config SEM_VALUE_MAX
 	help
 	  Maximum semaphore count in POSIX compliant Application.
 
+config PTHREAD_DYNAMIC_STACK
+	bool "Support for dynamic stacks"
+	default y
+	help
+	  POSIX 1003.1 allows a NULL pthread_attr_t* to be passed to
+	  pthread_create(3). However, Zephyr has traditionally required
+	  that the caller statically allocate a stack and pass it in via the
+	  pthread_attr_t*. With this option selected, NULL will be permitted
+	  and a suitable stack will be automatically allocated and assigned,
+	  inheriting permissions from the calling thread.
+
+if PTHREAD_DYNAMIC_STACK
+config PTHREAD_DYNAMIC_STACK_SIZE
+	int "Dynamic pthread stack size (in bytes)"
+	default 2048
+	range 1 4096
```

Comment:
I don't think that "1" is realistic lower bound here. The standard (https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_attr_getstack.html) says:

> The stacksize shall be at least {PTHREAD_STACK_MIN}.

Reply:
And yeah, why 4096 is upper bound?

Reply:
@pfalcon - the only reference I have in Zephyr for `PTHREAD_STACK_MIN` is in `tests/posix/eventfd/src/main.c`. Is there something analogous for Zephyr threads? All I can see is `XT_STACK_MIN_SIZE` for xtensa only. @andrewboie - any ideas?